hudi-utilities/src/main/java/org/apache/hudi/utilities/config/KafkaSourceConfig.java

```diff
@@ -63,6 +63,14 @@ public class KafkaSourceConfig extends HoodieConfig {
       .defaultValue(5000000L)
       .withDocumentation("Maximum number of records obtained in each batch.");
 
+  public static final ConfigProperty<Long> MAX_EVENTS_PER_KAFKA_PARTITION = ConfigProperty
```

Comment:
is there a way to simplify this to a fraction or do sth automatically? I just worry setting a total max (MAX_EVENTS_FROM_KAFKA_SOURCE) and then a per partition split limit can be a bit hard to set for the average user? thoughts? I ll think of concrete suggestions as well

Reply:
Yes,  Can we just have a parallelism config for Kafka read with default 0 indicating number of kafka partitions be used as parallelism "P". Instead of the current approach where one kafka partition translates to one spark partition, we can bin-pack the offsets into the "P" buckets with the constraint that one spark partition reads from only one kafka partition. Let me know if this makes sense.

Reply:
I have considered using a configuration similar to [`minPartitions`](https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html) in Spark Structured Streaming to control the concurrency, but this parameter does not intuitively control the data volume of each partition. Since my initial intention for this feature was to control the maximum data volume of each partition in order to reduce the time spent pulling data from Kafka, and considering the coherence of the configuration along with the config(**MAX_EVENTS_FROM_KAFKA_SOURCE**) of controlling the maximum data volume from Kafka, that's why I used the configuration for maximum data volume per partition.